packages/autocomplete-js/src/autocomplete.ts

```diff
@@ -50,6 +61,35 @@ export function autocomplete<TItem extends BaseItem>(
     createAutocomplete<TItem>({
       ...props.value.core,
       onStateChange(params) {
+        if (typeof insightsEnabled === 'undefined') {
+          // Get analytics status for algolia sources or `undefined` for non algolia sources
+          const analyticsStatuses = params.state.collections
+            .map(
+              (collection) =>
+                (collection.items[0]
+                  ?.__autocomplete_algoliaResultsMetadata as any)?.analytics
+            )
+            .filter((status) => typeof status !== 'undefined');
+
+          if (analyticsStatuses.length > 0) {
+            // Set insightsEnabled to true if at least one source has analytics enabled
+            insightsEnabled = analyticsStatuses.some(Boolean);
+          }
+
+          if (insightsEnabled) {
+            // Add plugin and refresh
+            update({
+              plugins: [...(options.plugins || []), createMockInsightsPlugin()],
+            });
+
+            setTimeout(() => {
+              // @ts-ignore
+              autocomplete.value.setIsOpen(true);
+              dom.value.input.focus();
+            });
```

Comment:
Not sure how to keep focus and state after a refresh without this hack.

Reply:
That's my guess yes

Reply:
It's annoying, but not a dealbreaker IMO, happy to validate we investigated this approach